tokio-trace-futures/examples/proxy_server.rs

```diff
@@ -0,0 +1,149 @@
+/*
+This example has been taken and modified from here :
+https://raw.githubusercontent.com/tokio-rs/tokio/master/tokio/examples/proxy.rs
+*/
+
+extern crate futures;
+extern crate tokio;
+#[macro_use]
+extern crate tokio_trace;
+extern crate tokio_trace_fmt;
+extern crate tokio_trace_futures;
+
+use tokio_trace::field;
+use tokio_trace_futures::Instrument;
+
+use std::env;
+use std::io::{self, Read, Write};
+use std::net::{Shutdown, SocketAddr};
+use std::sync::{Arc, Mutex};
+
+use tokio::io::{copy, shutdown};
+use tokio::net::{TcpListener, TcpStream};
+use tokio::prelude::*;
+
+fn main() -> Result<(), Box<std::error::Error>> {
+    let listen_addr = env::args().nth(1).unwrap_or("127.0.0.1:8081".to_string());
+    let listen_addr = listen_addr.parse::<SocketAddr>()?;
+
+    let server_addr = env::args().nth(2).unwrap_or("127.0.0.1:3000".to_string());
+    let server_addr = server_addr.parse::<SocketAddr>()?;
+
+    // Create a TCP listener which will listen for incoming connections.
+    let socket = TcpListener::bind(&listen_addr)?;
+    println!("Listening on: {}", listen_addr);
+    println!("Proxying to: {}", server_addr);
+
+    let done = socket
+        .incoming()
+        .map_err(|e| debug!(msg = "error accepting socket", error = field::display(e)))
+        .for_each(move |client| {
+            let server = TcpStream::connect(&server_addr);
+            let mut client_addr: Option<SocketAddr> = None;
```

Comment:
This type annotation is fine, but it shouldn't be strictly necessary.
```suggestion
            let mut client_addr = None;
```